### ISSUE TITLE
Paragraph draws incorrectly when drawing 2 chars, then 1 char on the first line

### ISSUE THREAD

**Issue by kirbycool** [fdehau/tui-rs#379]
**Describe the bug**
Hi. Thanks for the great library!
 I'm running into some weird behavior specifically when drawing two characters on the first line, then one character with the `Paragraph` widget; I end up with only the second character from the first draw and no characters from the second draw.

**To Reproduce**
Here's a minimal repro with Termion, drawing "ab", then "a", and waiting on a keypress to exit.
```
termion = "1.5.5"
tui = "0.10.0"
```
```rust
use std::io;
use termion::input::TermRead;
use termion::raw::IntoRawMode;
use tui::backend::TermionBackend;
use tui::text::Text;
use tui::widgets::Paragraph;
use tui::Terminal;

fn main() {
    let stdout = io::stdout().into_raw_mode().unwrap();
    let backend = TermionBackend::new(stdout);
    let mut terminal = Terminal::new(backend).unwrap();

    terminal
        .draw(|f| {
            let p = Paragraph::new(Text::from("ab"));
            f.render_widget(p, f.size());
        })
        .unwrap();

    terminal
        .draw(|f| {
            let p = Paragraph::new(Text::from("a"));
            f.render_widget(p, f.size());
        })
        .unwrap();

    io::stdin().keys().next();
}
```

**Expected behavior**
Ouput should contain the last draw call
```
a
```

**Actual behavior**
Output looks like this (`<space>b`)
```
 b
```
![image](https://user-images.githubusercontent.com/696150/93705894-1629b980-fad6-11ea-9c88-66fb5aa464d5.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 
 - Terminal Emulator: Terminal
 - Font: Source Code Pro
 - Crate: 0.10.0
 - Backend: termion

**Additional context**
I do get the expected behavior if I change the inputs to `"\nab"` and `"\na"` so that the text is not on the first line.
```

a
```
I'm not super familiar with tty details, so apologies if this is a problem deeper in the backend. I'm happy to spend time digging into this if you think it's a bug.

**Comment by fdehau** [comment]
This should have been fixed in #347. Could you confirm that upgrading tui to 0.11.0 fix your issue ?

**Comment by kirbycool** [comment]
Looks solved with 0.11.0. Thanks!
